integration/kotlinx-coroutines-guava/src/ListenableFuture.kt

```diff
@@ -70,10 +70,21 @@ public fun <T> CoroutineScope.future(
  * When `this` `ListenableFuture` is [successfully cancelled][java.util.concurrent.Future.cancel],
  * it will cancel the returned `Deferred`.
  *
- * When the returned `Deferred` is [cancelled][Deferred.cancel()], it will try to propagate the
+ * When the returned `Deferred` is [cancelled][Deferred.cancel], it will try to propagate the
  * cancellation to `this` `ListenableFuture`. Propagation will succeed, barring a race with the
  * `ListenableFuture` completing normally. This is the only case in which the returned `Deferred`
  * will complete with a different outcome than `this` `ListenableFuture`.
+ *
+ * Note: Cancellation of the coroutine that calls [await][Deferred.await] on the returned
+ * deferred, **does not** cancel [Deferred] and original [ListenableFuture].
+ * To propagate cancellation from awaiting coroutine to the [ListenableFuture], use [ListenableFuture.await].
+ *
+ * ```
+ * suspend fun foo() {
+ *   listenableFuture.asDeferred().await()  // Doesn't propagate `foo`'s cancellation to the `listenableFuture`
```

Comment:
We're in the weeds on semantics, but `foo` isn't cancellable, because it's a function. The example isn't quite complete without a coroutine invoking `foo`. Using a `launch` coroutine builder might be easier to write technically correct comments on, when explaining calling-coroutine cancellation.